node/src/services/blockchain/transaction.rs

```diff
@@ -0,0 +1,150 @@
+use std::{
+    str::FromStr,
+    time::{Duration, Instant},
+};
+
+use anyhow::anyhow;
+use log::*;
+use sp_core::H256;
+use storage_hub_infra::actor::ActorHandle;
+use tokio::sync::mpsc::Receiver;
+
+use crate::services::blockchain::{commands::BlockchainServiceInterface, types::ExtrinsicResult};
+
+use super::{types::ExtrinsicHash, BlockchainService};
+
+const LOG_TARGET: &str = "blockchain-transaction";
+
+#[derive(Debug)]
+pub struct SubmittedTransaction {
+    watcher: Receiver<String>,
+    hash: ExtrinsicHash,
+    timeout: Option<Duration>,
+}
+
+const NO_TIMEOUT_WARN: Duration = Duration::from_secs(60);
+
+impl SubmittedTransaction {
+    pub fn new(watcher: Receiver<String>, hash: H256) -> Self {
+        Self {
+            watcher,
+            hash,
+            timeout: None,
+        }
+    }
+
+    pub fn with_timeout(mut self, timeout: Duration) -> Self {
+        self.timeout = Some(timeout);
+        self
+    }
+
+    pub async fn watch_for_success(
```

Comment:
```suggestion
    /// Handles the lifecycle of a submitted transaction.
    ///
    /// Waits for the transaction to be included in a block.
    /// If the transaction is not included in a block within the specified timeout, it will be
    /// considered failed and an error will be returned.
    pub async fn watch_for_success(
```